### PR TITLE
Add fab to the member list screen to invite users.

### LINKED ISSUE #2226: Add FAB to room members list
And remove the corresponding menu item.

Visibility is the same than for the menu item.

When visible, I think it's OK that the FAB does not hide automatically when scrolling.

image

Figma link: https://www.figma.com/file/cDsGCH7Cz9Od5xHkn2dxir/Add-people-to-room?node-id=20%3A0

### PR DESCRIPTION
**Pull request by onurays**
Fixes #2226 

Screenshot_1602498345

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/roomprofile/members/RoomMemberListFragment.kt [hunk 1]
```diff
@@ -49,38 +49,44 @@ class RoomMemberListFragment @Inject constructor(
...
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         roomMemberListController.callback = this
         setupToolbar(roomSettingsToolbar)
         setupSearchView()
+        setupInviteUsersButton()
         recyclerView.configureWith(roomMemberListController, hasFixedSize = true)
+        // Hide FAB when list is scrolling
+        recyclerView.addOnScrollListener(
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/roomprofile/members/RoomMemberListFragment.kt:62]: Move that code to `setupInviteUsersButton()`?

File: vector/src/main/res/layout/fragment_room_member_list.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        android:layout_height="wrap_content"
+        android:layout_marginEnd="16dp"
+        android:layout_marginBottom="16dp"
+        android:contentDescription="@string/add_members_to_room"
+        android:scaleType="center"
+        android:src="@drawable/ic_fab_add_members"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:maxImageSize="36dp"
+        tools:visibility="visible" />
```
Comment (bmarty) [inline vector/src/main/res/layout/fragment_room_member_list.xml:22]: Please remove, this is the default visibility

File: vector/src/main/res/layout/fragment_room_member_list.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (bmarty) [inline vector/src/main/res/layout/fragment_room_member_list.xml:2]: A FrameLayout is enough to do this, with `android:layout_gravity="bottom|end"` for the FAB

File: vector/src/main/java/im/vector/app/features/roomprofile/members/RoomMemberListFragment.kt [hunk 4]
```diff
@@ -131,5 +137,6 @@ class RoomMemberListFragment @Inject constructor(
             roomSettingsToolbarTitleView.text = it.displayName
             avatarRenderer.render(it.toMatrixItem(), roomSettingsToolbarAvatarImageView)
         }
+        inviteUsersButton.isVisible = state.actionsPermissions.canInvite
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/roomprofile/members/RoomMemberListFragment.kt:140]: This is not really the room summary rendering.
I would have add this line of code at the end of `invalidate()` or by creating a new dedicated method.
Also what will happen if the state is updated while scrolling? (this is a minor bug, though)

**Review by bmarty — CHANGES_REQUESTED** [review]
A few remarks, else LGTM

File: vector/src/main/java/im/vector/app/features/roomprofile/members/RoomMemberListFragment.kt (hunk 1 above)
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/roomprofile/members/RoomMemberListFragment.kt:71]: Done.

File: vector/src/main/res/layout/fragment_room_member_list.xml (hunk 2 above)
Comment (onurays) [reply vector/src/main/res/layout/fragment_room_member_list.xml]: Done.

File: vector/src/main/res/layout/fragment_room_member_list.xml (hunk 3 above)
Comment (onurays) [reply vector/src/main/res/layout/fragment_room_member_list.xml]: Cool.

File: vector/src/main/java/im/vector/app/features/roomprofile/members/RoomMemberListFragment.kt (hunk 4 above)
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/roomprofile/members/RoomMemberListFragment.kt]: Done. I think RecyclerView dragging event will be triggered quickly and probably it won't be visible at all.

File: vector/src/main/java/im/vector/app/features/roomprofile/members/RoomMemberListFragment.kt [hunk 5]
```diff
@@ -49,38 +49,44 @@ class RoomMemberListFragment @Inject constructor(
...
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         roomMemberListController.callback = this
         setupToolbar(roomSettingsToolbar)
         setupSearchView()
+        setupInviteUsersButton()
         recyclerView.configureWith(roomMemberListController, hasFixedSize = true)
         viewModel.selectSubscribe(this, RoomMemberListViewState::actionsPermissions) {
             invalidateOptionsMenu()
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/roomprofile/members/RoomMemberListFragment.kt:62]: This is not necessary anymore then

Comment (bmarty) [reply vector/src/main/java/im/vector/app/features/roomprofile/members/RoomMemberListFragment.kt:62]: (I will do the cleanup later)

**Pull request merged**